src/main/java/uk/gov/hmcts/ccd/domain/service/accessprofile/filter/matcher/BeginDateEndDateMatcher.java

```diff
@@ -0,0 +1,30 @@
+package uk.gov.hmcts.ccd.domain.service.accessprofile.filter.matcher;
+
+import java.time.Instant;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleAssignment;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleAssignmentFilteringResult;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleMatchingResult;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+
+@Slf4j
+@Component
+public class BeginDateEndDateMatcher implements AttributeMatcher {
+
+    @Override
+    public boolean matchAttribute(RoleAssignmentFilteringResult result, CaseDetails caseDetails) {
+        RoleAssignment roleAssignment = result.getRoleAssignment();
+        log.debug("Apply filter on start {} and end time {} for role assignment {}",
+            roleAssignment.getBeginTime(),
+            roleAssignment.getEndTime(),
+            roleAssignment.getId());
+        RoleMatchingResult matchingResult = result.getRoleMatchingResult();
+        if (roleAssignment.getBeginTime() != null && roleAssignment.getEndTime() != null) {
+            Instant now = Instant.now();
```

Comment:
can you use https://github.com/hmcts/ccd-data-store-api/blob/master/src/main/java/uk/gov/hmcts/ccd/CoreCaseDataApplication.java#L41 please

Reply:
we are going with Instant, had a chat with Rafal on this